blocks/init/src/Blocks/components/load-more/assets/load-more.js

```diff
@@ -0,0 +1,322 @@
+
+		// Get query var.
+		this.QUERY_VAR = 'load-more-page';
+
+		// Set internal store.
+		this.data = {};
+	}
+
+	/**
+	 * Store return object by ID.
+	 *
+	 * @param {string} id Trigger unique ID.
+	 *
+	 * @returns {object} Object of data based on the ID.
+	 */
+	getData(id) {
+		console.log(typeof id);
+		
+		return this.data[id];
+	}
+
+	/**
+	 * Set top level initial store.
+	 *
+	 * @param {string} id Trigger unique ID.
+	 * @param {object} value Object of data to store.
+	 */
+	setDataTop(id, value) {
+		this.data = {
+			...this.data,
+			[id]: value,
+		};
+	}
+
+	/**
+	 * Set store by ID and parent object key.
+	 *
+	 * @param {string} id Trigger unique ID.
+	 * @param {string} parent Key name in the object.
+	 * @param {mixed} value Value stored to the key in the object.
+	 */
+	setData(id, parent, value) {
+		this.data[id] = {
+			...this.data[id],
+			[parent]: value,
+		};
+	}
+
+	/**
+	 * Init all event-listeners.
+	 */
+	init() {
+		// Loop all triggers.
+		[...this.triggerElements].forEach((trigger) => {
+
+			// Get trigger ID.
+			const id = trigger.getAttribute(this.ATTR_ID);
+
+			// Prepare data.
+			this.prepareData(trigger, id);
+
+			// Get container.
+			const container = document.querySelector(`[${this.ATTR_ID}="${id}"]`);
+
+			// Fetch data from url param.
+			this.fetchDataFromUrl(trigger, id, container);
+
+			// Set store to prevent corrections after initial load.
+			this.setData(id, 'urlLoadMoreAction', false);
+
+			// Add listener on trigger click.
+			trigger.addEventListener('click', (event) => {
+				event.preventDefault();
+
+				// Prevent clicking on the button until loading state is active.
+				if (!event.target.classList.contains(this.CLASS_IS_LOADING)) {
+					// Fetch new data.
+					this.fetchData(trigger, id, container);
+				}
+			});
+		});
+	}
+
+	/**
+	 * Check per page details.
+	 *
+	 * @param {object} trigger Element for trigger.
+	 * @param {string} id Trigger unique ID.
+	 *
+	 * @returns {void}
+	 */
+	prepareData = (trigger, id) => {
+		const initialItems = trigger.getAttribute(this.ATTR_INITIAL_ITEMS);
+		const query = trigger.getAttribute(this.ATTR_QUERY);
+		const type = trigger.getAttribute(this.ATTR_TYPE);
+		const perPageOverride = trigger.getAttribute(this.ATTR_PER_PAGE_OVERRIDE);
+
+		// Set store top level.
+		this.setDataTop(id, {
+			action: this.ajaxHandler,
+			query,
+			initialItems,
+			id,
+			type,
+			perPageOverride,
+			urlLoadMoreAction: false,
+			preventAction: false,
+		});
+	}
+
+	/**
+	 * Fetch new data.
+	 *
+	 * @param {object} trigger Element for trigger.
+	 * @param {string} id Trigger unique ID.
+	 * @param {object} container Element for container.
+	 *
+	 * @returns {void}
+	 */
+	fetchData = (trigger, id, container) => {
+
+		// Get data from store.
+		const data = this.getData(id);
+
+		// Bailout if there are no more actions.
+		if (data.preventAction) {
+			return;
+		}
+
+		// Prepare body data.
+		const body = {
+			method: 'POST',
```

Comment:
_Fetching_ implies `GET`, not `POST` 😬 

We should use appropriate HTTP methods for things we do with the API (or AJAX in this case).

Reply:
The problem here is that I send the data from frontend to backend via json format because we have multiple decoded params that can't be sent via get parameter. So this is why I was using POST

Reply:
I think we discussed this yesterday, no?